### PR TITLE
Allow votes from the user frontend

### PR DESCRIPTION
**Pull request by kleingeist**
This PR includes the changes from #273 
API changes depend on liqd/adhocracy4#89

It is missing
* Changes to the api.js in core
* logic for the AJAX request
  * what to show during the request
  * how to handle/show errors/success
* Layout for the management form

### CODE REVIEW

**Comment by 2e2a** [comment]
api and serializer lgtm

File: apps/polls/assets/QuestionForm.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      <section>
+        <div className="commenting">
+          <div className="commenting__content">
+            <div className="form-group">
+              <label
+                htmlFor={'id_questions-' + this.props.key + '-name'}>
+                {django.gettext('Question:')}
+              </label>
+              <input
+                className="form-control"
```
Comment (vellip) [inline apps/polls/assets/QuestionForm.jsx]: `.form-control` is of no importance anymore, we removed it. I know, theoretically the same goes for documents.

File: apps/polls/assets/PollManagement.jsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+    let data = {
+      questions: this.state.questions
+    }
+
+    let promise = api.poll.change(data, this.props.poll.id)
+    promise
+      .done(function (data) {
+        this.setState({
+          successMessage: django.gettext('The poll has been updated.')
```
Comment (vellip) [inline apps/polls/assets/PollManagement.jsx]: You'll want to reset previous errors here.

File: apps/polls/assets/PollManagement.jsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+        setTimeout(function () {
+          this.setState({
+            successMessage: ''
+          })
+        }.bind(this), 1500)
+      }.bind(this))
+      .fail(function (xhr, status, err) {
+        this.setState({
+          errors: xhr.responseJSON.questions || []
```
Comment (vellip) [inline apps/polls/assets/PollManagement.jsx]: Why a fallback? If there are no errors to set (which `errors: []` would mean) you don't want to call `setState`.

File: apps/polls/assets/PollManagement.jsx [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        this.setState({
+          successMessage: django.gettext('The poll has been updated.')
+        })
+
+        setTimeout(function () {
+          this.setState({
+            successMessage: ''
+          })
+        }.bind(this), 1500)
+      }.bind(this))
```
Comment (vellip) [inline apps/polls/assets/PollManagement.jsx]: Consider using [fat arrow functions](https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Functions#Arrow_functions) to get rid of all the `.bind`s

File: apps/polls/assets/Question.jsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+  handleSubmit: function (event) {
+    event.preventDefault()
+
+    let newChoice = this.state.selectedChoice
+
+    let submitData = {
+      choice: this.props.question.choices[newChoice].id
+    }
+
+    return api.poll.vote(submitData, this.props.question.id)
```
Comment (vellip) [inline apps/polls/assets/Question.jsx]: No need to return the promise.

File: apps/polls/migrations/0002_auto_20170503_1524.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
```
Comment (kleingeist) [inline apps/polls/migrations/0002_auto_20170503_1524.py]: i'll leave the generated title this time but will name it manually next time. 
maybe we could also squash them when the feature is ready

File: apps/polls/assets/PollManagement.jsx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+  */
+
+  getNextQuestionKey: function () {
+    /** Get an artifical key for non-commited questions.
+     *
+     *  Prefix to prevent collisions with real database keys;
+     */
+    var questionKey = 'local_' + (this.state.maxQuestionKey + 1)
+    this.setState({maxQuestionKey: this.state.maxQuestionKey + 1})
+    return questionKey
```
Comment (kleingeist) [inline apps/polls/assets/PollManagement.jsx]: good point, i changed it to be a class variable. the changes should be reflected in documents, too

**Pull request merged**